x-pack/plugins/fleet/public/components/platform_selector.tsx

```diff
@@ -50,29 +51,22 @@ export const PlatformSelector: React.FunctionComponent<Props> = ({
   linuxDebCommand,
   linuxRpmCommand,
   k8sCommand,
-  isK8s,
+  hasK8sIntegration,
   isManaged,
   enrollToken,
-  isFleet,
+  hasFleetServer,
   fullCopyButton,
   onCopy,
 }) => {
   const { platform, setPlatform } = usePlatform();
 
   useEffect(() => {
-    setPlatform(isK8s ? 'kubernetes' : 'linux');
-  }, [isK8s, isManaged, setPlatform]);
+    setPlatform(hasK8sIntegration ? 'kubernetes' : 'linux');
+  }, [hasK8sIntegration, setPlatform]);
 
-  // In case of fleet server installation remove Kubernetes as platform option
-  const REDUCED_PLATFORM_OPTIONS = [...PLATFORM_OPTIONS];
-  const indexOfK8s = REDUCED_PLATFORM_OPTIONS.findIndex((object) => {
-    return object.id === 'kubernetes';
-  });
-  if (indexOfK8s !== -1) {
-    REDUCED_PLATFORM_OPTIONS.splice(indexOfK8s, 1);
-  }
-  const useReduce = isFleet || (!isManaged && !isK8s) ? true : false;
-  const options = useReduce ? REDUCED_PLATFORM_OPTIONS : PLATFORM_OPTIONS;
+  // In case of fleet server installation or standalone agent without
+  // Kubernetes integration in the policy use reduced platform options
+  const useReduce = hasFleetServer || (!isManaged && !hasK8sIntegration) ? true : false;
```

Comment:
Nit: this could be `const useReduce = hasFleetServer || (!isManaged && !hasK8sIntegration)`